precompiles/staking/staking_test.go

```diff
@@ -238,1238 +173,379 @@ func Test_InvalidABI(t *testing.T) {
-	// 	err = sdkKeepers.BankKeeper.SendCoinsFromModuleToAccount(ctx, fungibletypes.ModuleName, staker, coins)
-	// 	require.NoError(t, err)
-
-	// 	stakerAddr := common.BytesToAddress(staker.Bytes())
-
-	// 	argsStake := []interface{}{
-	// 		stakerEthAddr,
-	// 		validatorSrc.OperatorAddress,
-	// 		coins.AmountOf(config.BaseDenom).BigInt(),
-	// 	}
-
-	// 	// stake to validator src
-	// 	stakeMethodID := abi.Methods[StakeMethodName]
-	// 	_, err = contract.Stake(ctx, mockEVM, &vm.Contract{CallerAddress: stakerAddr}, &stakeMethodID, argsStake)
-	// 	require.NoError(t, err)
-
-	// 	argsMoveStake := []interface{}{stakerEthAddr, validatorSrc.OperatorAddress, validatorDest.OperatorAddress}
-
-	// 	// ACT
-	// 	_, err = contract.MoveStake(ctx, mockEVM, &vm.Contract{CallerAddress: stakerAddr}, &methodID, argsMoveStake)
-
-	// 	// ASSERT
-	// 	require.Error(t, err)
-	// })
-
-	// t.Run("should fail if caller is not staker", func(t *testing.T) {
-	// 	// ARRANGE
-	// 	ctx, contract, abi, sdkKeepers, mockEVM, mockVMContract := setup(t)
-	// 	methodID := abi.Methods[MoveStakeMethodName]
-	// 	r := rand.New(rand.NewSource(42))
-	// 	validatorSrc := sample.Validator(t, r)
-	// 	sdkKeepers.StakingKeeper.SetValidator(ctx, validatorSrc)
-	// 	validatorDest := sample.Validator(t, r)
-	// 	sdkKeepers.StakingKeeper.SetValidator(ctx, validatorDest)
-
-	// 	staker := sample.Bech32AccAddress()
-	// 	stakerEthAddr := common.BytesToAddress(staker.Bytes())
-	// 	coins := sample.Coins()
-	// 	err := sdkKeepers.BankKeeper.MintCoins(ctx, fungibletypes.ModuleName, sample.Coins())
-	// 	require.NoError(t, err)
-	// 	err = sdkKeepers.BankKeeper.SendCoinsFromModuleToAccount(ctx, fungibletypes.ModuleName, staker, coins)
-	// 	require.NoError(t, err)
-
-	// 	stakerAddr := common.BytesToAddress(staker.Bytes())
-	// 	mockVMContract.CallerAddress = stakerAddr
-	// 	argsStake := []interface{}{
-	// 		stakerEthAddr,
-	// 		validatorSrc.OperatorAddress,
-	// 		coins.AmountOf(config.BaseDenom).BigInt(),
-	// 	}
-
-	// 	// stake to validator src
-	// 	stakeMethodID := abi.Methods[StakeMethodName]
-	// 	mockVMContract.Input = packInputArgs(t, stakeMethodID, argsStake...)
-	// 	_, err = contract.Run(mockEVM, mockVMContract, false)
-	// 	require.NoError(t, err)
-
-	// 	argsMoveStake := []interface{}{
-	// 		stakerEthAddr,
-	// 		validatorSrc.OperatorAddress,
-	// 		validatorDest.OperatorAddress,
-	// 		coins.AmountOf(config.BaseDenom).BigInt(),
-	// 	}
-	// 	mockVMContract.Input = packInputArgs(t, methodID, argsMoveStake...)
-
-	// 	callerEthAddr := common.BytesToAddress(sample.Bech32AccAddress().Bytes())
-	// 	mockVMContract.CallerAddress = callerEthAddr
-
-	// 	// ACT
-	// 	_, err = contract.Run(mockEVM, mockVMContract, false)
-
-	// 	// ASSERT
-	// 	require.ErrorContains(t, err, "caller is not staker")
-	// })
+	// Intiliaze codecs and gas config.
+	var encoding ethermint.EncodingConfig
+	appCodec := encoding.Codec
+	gasConfig := storetypes.TransientGasConfig()
```

Comment:
Why do we need to use TransientGasConfig, instead of KVGasConfig .

Reply:
Not sure about this one, this test is outside scope and it was developed a while ago. I'll investigate it soon.